src/s1reader/s1_info.py

```diff
@@ -48,27 +56,109 @@ def _plot_bursts(safe_path, output_dir="burst_maps"):
     plot_bursts.burst_map(safe_path, orbit_dir, xs, ys, epsg, output_filename)
 
 
+def get_frame_bounds(safe_path: Union[Path, str]) -> List[float]:
+    """Get the bounding box of the frame from the union of all burst bounds.
+
+    bounding box format is [lonmin, latmin, lonmax, latmax]
+
+    Notes
+    -----
+    Will use the preview/map-overlay.kml file if it exists, otherwise will
+    use the union of all burst bounds (which is slower).
+
+    Parameters
+    ----------
+    safe_path : Union[Path, str]
+        Path to the SAFE directory or zip file.
+
+    Returns
+    -------
+    List[float]
+        [lonmin, latmin, lonmax, latmax]
+    """
+    try:
+        return _bounds_from_preview(safe_path)
+    except Exception as e:
+        warnings.warn(f"Could not get bounds for {safe_path}: {e}")
+    return _bounds_from_bursts(safe_path)
+
+
+def _bounds_from_preview(safe_path: Union[Path, str]) -> List[float]:
+    """Get the bounding box of the frame from the preview/map-overlay.kml."""
+    # looking for:
+    # S1A_IW_SLC__1SDV_20221005T125539_20221005T125606_045307_056AA5_CB45.SAFE/preview/map-overlay.kml
+    if _is_safe_dir(safe_path):
+        overlay_path = Path(safe_path) / "preview" / "map-overlay.kml"
+        temp_dir = None
+    else:
+        # The name of the unzipped .SAFE directory (with .zip stripped)
+        temp_dir = tempfile.mkdtemp()
+        with zipfile.ZipFile(safe_path, "r") as zip_ref:
+            zname = [
+                zi
+                for zi in zip_ref.infolist()
+                if "preview/map-overlay.kml" in zi.filename
+            ]
+            overlay_path = Path(zip_ref.extract(zname[0], path=temp_dir))
+
+    # Check that they have all the necessary kmls
+    if not overlay_path.exists():
+        raise ValueError(f"{overlay_path} does not exist.")
+    root = ET.parse(overlay_path).getroot()
+
+    # point_str looks like:
+    # <coordinates>-102.552971,31.482372 -105.191353,31.887299...
+    point_str = list(elem.text for elem in root.iter("coordinates"))[0]
+    coords = [p.split(",") for p in point_str.split()]
+    lons, lats = zip(*[(float(lon), float(lat)) for lon, lat in coords])
+    if temp_dir is not None:
+        shutil.rmtree(temp_dir)
+    return [min(lons), min(lats), max(lons), max(lats)]
+
+
+def _bounds_from_bursts(safe_path: Union[Path, str]) -> List[float]:
+    """Get the bounding box of the frame from the union of all burst bounds."""
+    # Get all the bursts from subswath 1, 2, 3
+    bursts = get_bursts(safe_path)
```

Comment:
This comment would be relevant only if you want to make this code to work globally.

`get_bursts()` does provide option to choose polarization, but this line will only try to load VV. This may cause an issue for Cryosphere data (Greenland or Antarctica) because the polarization there is usually SDH (HH/HV).

Reply:
> This comment would be relevant only if you want to make this code to work globally.

Sorry, I'm not sure what you mean by that? 



> get_bursts() does provide option to choose polarization, but this line will only try to load VV

Good catch, I'll add a loop to try all polarizations and break on the first one we find.

Reply:
What I meant by "globally" was all Sentinel-1 data. Not all data has VV polarization as I mentioned above.
I think it would be smarter way to choose the polarization from SAFE string, but looping through the polarization would be fair enough.